crates/grafbase-sdk/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+#![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/README.md"))]
+#![deny(missing_docs)]
+
+#[doc(hidden)]
+pub mod extension;
+pub mod host_io;
+pub mod types;
+
+pub use extension::{Extension, Resolver};
+pub use wit::{Error, ExtensionType, SharedContext};
+
+use semver::Version;
+use std::sync::LazyLock;
+
+/// The minimum version of Grafbase Gateway that can run this SDK.
+pub const MINIMUM_HOST_VERSION: Version = Version::new(0, 48, 0);
+
+/// The version of the SDK.
+pub static GUEST_VERSION: LazyLock<Version> =
+    LazyLock::new(|| Version::parse(std::env!("CARGO_PKG_VERSION")).expect("must be valid semver"));
+
+#[doc(hidden)]
+pub fn check_host_version(host_version: u32) -> i32 {
+    let version = unpack_version(host_version);
+
+    if version < MINIMUM_HOST_VERSION {
+        -(pack_version(&MINIMUM_HOST_VERSION) as i32)
+    } else {
+        pack_version(&GUEST_VERSION) as i32
+    }
+}
+
+fn pack_version(version: &Version) -> u32 {
+    let major = version.major as u32;
```

Comment:
You should mask the relevant bits before shifting, so `let major = (major & 0xFF) << 16`
I would strongly consider using an`i64` instead of `i32`, You could use 20 bits per number for this. It's far from unheard to have versions in the hundreds. Otel-collector is at `0.118` for example.

Reply:
A back and forth unit test would be nice also :D

Reply:
I forgot WASM supports 64-bit integers, only the pointers are 32-bit... Yeah will do.

Reply:
nice idea for the version though